src/features/defi/providers/osmosis/components/OsmosisManager/Withdraw/components/Withdraw.tsx

```diff
@@ -0,0 +1,534 @@
+      if (
+        !(osmosisOpportunity && state && state.opportunity && underlyingAsset0 && underlyingAsset1)
+      ) {
+        return undefined
+      }
+
+      const { assetReference: poolAssetReference } = fromAssetId(state.opportunity.assetId)
+      const id = getPoolIdFromAssetReference(poolAssetReference)
+      if (!id) return undefined
+
+      if (!poolData) setPoolData(await getPool(id))
+
+      if (
+        !(
+          poolData &&
+          poolData.pool_assets &&
+          poolData.total_shares &&
+          poolData.total_weight !== '0'
+        )
+      ) {
+        return undefined
+      }
+
+      /**
+       * pool_share_fraction = input_pool_shares / total_pool_shares
+       * This represents the fraction of the total number of pool LP tokens being redeemed
+       */
+      if (bnOrZero(poolData.total_shares.amount).eq(bn(0))) return undefined
+      const poolShareFraction = bnOrZero(inputPoolSharesBaseUnit)
+        .dividedBy(poolData.total_shares.amount)
+        .toString()
+
+      /**
+       * token_out_n_amount = token_n_amount * pool_share_fraction * asset_n_proportional_weight
+       * This represents the number of each underlying token being returned to the user
+       */
+      const tokenOut0AmountBaseUnit = bnOrZero(poolData.pool_assets[0].token.amount)
+        .multipliedBy(poolShareFraction)
+        .toFixed(0)
+        .toString()
+      const tokenOut1AmountBaseUnit = bnOrZero(poolData.pool_assets[1].token.amount)
+        .multipliedBy(poolShareFraction)
+        .toFixed(0)
+        .toString()
+
+      const asset0Reference = fromAssetId(underlyingAsset0.assetId).assetReference
+      const asset1Reference = fromAssetId(underlyingAsset1.assetId).assetReference
+
+      return [
+        {
+          amount: tokenOut0AmountBaseUnit,
+          denom:
+            asset0Reference === ASSET_REFERENCE.Osmosis
+              ? 'uosmo'
+              : `${ASSET_NAMESPACE.ibc}/${asset0Reference}`,
+        },
+        {
+          amount: tokenOut1AmountBaseUnit,
+          denom:
+            asset1Reference === ASSET_REFERENCE.Osmosis
+              ? 'uosmo'
+              : `${ASSET_NAMESPACE.ibc}/${asset1Reference}`,
+        },
+      ]
+    },
+    [osmosisOpportunity, poolData, state, underlyingAsset0, underlyingAsset1],
+  )
+
+  const handleInputChange = (value: string, isFiat?: boolean) => {
+    if (!(poolAssetMarketData && lpAsset)) return
+    const divisor = bnOrZero(poolAssetMarketData.price)
+
+    /**
+     * Lol at this double-ternary operation.
+     * If this is a fiat value, we can get the crypto amount by diving value by price,
+     * but we have to check first if the price is zero so we don't divide by zero.
+     * If price is zero, just set _value to '0'. Otherwise, set _value to value/price.
+     */
+    const _value = isFiat
+      ? divisor.isGreaterThan(0)
+        ? bnOrZero(value).dividedBy(divisor).toString()
```

Comment:
If we opt chain on 0 being falsy, that will be a much cleaner way, avoiding the nested ternary
```suggestion
        bnOrZero(value).dividedBy(divisor.toFixed() ?? 1).toString()
```